jobs/api/CommuteSearchSample.cs

```diff
@@ -0,0 +1,101 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Google.Apis.JobService.v2;
+using Google.Apis.JobService.v2.Data;
+
+namespace Google.Samples
+{
+
+    /// <summary>
+    /// The samples in this file introduce how to do a commute search.
+    ///
+    /// Note: Commute Search is different from location search. It only take latitude and longitude as
+    /// the start location.
+    /// </summary>
+    public static class CommuteSearchSample
+    {
+
+        private static JobServiceService jobService = JobServiceUtils.JobService;
+
+        // [START commute_search]
+
+        public static void CommuteSearch(string companyName)
+        {
+            // Make sure to set the requestMetadata the same as the associated search request
+            RequestMetadata requestMetadata =
+                new RequestMetadata
+                {
+                    // Make sure to hash your userID
+                    UserId = "HashedUserId",
+                    // Make sure to hash the sessionID
+                    SessionId = "HashedSessionID",
+                    // Domain of the website where the search is conducted
+                    Domain = "www.google.com",
+                };
+            JobQuery jobQuery =
+                new JobQuery
+                {
+                    CommuteFilter =
+                        {
+                            RoadTraffic = "TRAFFIC_FREE",
+                            Method = "TRANSIT",
+                            TravelTime = "1000s",
+                            StartLocation =
+                                {
+                                    Latitude = 37.422408,
+                                    Longitude = -122.085609
+                                }
+                        }
+                };
+            if (companyName != null)
+            {
+                jobQuery.CompanyNames = new List<string> { companyName };
+            }
+            SearchJobsRequest searchJobsRequest =
+                new SearchJobsRequest
+                {
+                    Query = jobQuery,
+                    RequestMetadata = requestMetadata,
+                    JobView = "FULL",
+                    EnablePreciseResultSize = true,
+                };
+            SearchJobsResponse response = jobService.Jobs.Search(searchJobsRequest).Execute();
+            Console.WriteLine(response);
+        }
+        // [END commute_search]
+
+        public static void Main(string[] args)
```

Comment:
I find many Main() functions, all in the same .csproj.  Which one actually runs?

The common pattern in this repo is to build a single binary (one Main function) and use https://github.com/GoogleCloudPlatform/dotnet-docs-samples/tree/master/commandlineutil to dispatch the command line.

Also, needs a README.md and tests.  See https://github.com/GoogleCloudPlatform/dotnet-docs-samples/tree/927526578de0644a09a1e6611c848bd4f78f7a65/monitoring/api/UptimeCheckSample for an example of a simple command line app, and https://github.com/GoogleCloudPlatform/dotnet-docs-samples/tree/927526578de0644a09a1e6611c848bd4f78f7a65/monitoring/api/UptimeCheckTest for its tests.